src/dsp_tools/utils/connection_live.py

```diff
@@ -395,5 +310,57 @@ def delete(
             headers=headers,
             timeout=timeout,
         )
-        check_for_api_error(response)
         return cast(dict[str, Any], response.json())
+
+    def _try_network_action(self, action: Callable[..., Any]) -> Any:
+        """
+        Try 7 times to execute a HTTP request.
+        If a timeout error, a ConnectionError, or a requests.RequestException occur,
+        or if the response indicates that there is a non-permanent server-side problem,
+        this function waits and retries the HTTP request.
+        The waiting times are 1, 2, 4, 8, 16, 32, 64 seconds.
+
+        Args:
+            action: a lambda with the code to be executed, or a function
+
+        Raises:
+            BaseError: if the action fails permanently
+            unexpected exceptions: if the action fails with an unexpected exception
+
+        Returns:
+            the return value of action
+        """
+        for i in range(7):
+            try:
+                response: requests.Response = action()
+            except (TimeoutError, ReadTimeout, ReadTimeoutError):
+                msg = f"Timeout Error: Try reconnecting to DSP server, next attempt in {2 ** i} seconds..."
+                print(f"{datetime.now()}: {msg}")
+                logger.error(f"{msg} (retry-counter {i=:})", exc_info=True)
+                time.sleep(2**i)
+                continue
+            except (ConnectionError, RequestException):
+                msg = f"Network Error: Try reconnecting to DSP server, next attempt in {2 ** i} seconds..."
+                print(f"{datetime.now()}: {msg}")
+                logger.error(f"{msg} (retry-counter {i=:})", exc_info=True)
+                time.sleep(2**i)
+                continue
+
+            in_500_range = 500 <= response.status_code < 600
+            try_again_later = "try again later" in response.text
+            if try_again_later or in_500_range:
+                msg = f"Transient Error: Try reconnecting to DSP server, next attempt in {2 ** i} seconds..."
+                print(f"{datetime.now()}: {msg}")
+                logger.error(f"{msg} (retry-counter {i=:})", exc_info=True)
+                time.sleep(2**i)
+                continue
+            elif response.status_code != 200:
+                raise BaseError(
+                    message="Permanently unable to execute the network action. See logs for more details.",
+                    status_code=response.status_code,
+                    json_content_of_api_response=response.text,
+                    reason_from_api=response.reason,
+                    api_route=response.url,
+                )
+            else:
+                return response
```

Comment:
```suggestion
            return response
```

no need to call else here, in any other code path, it either continues or raises.

Reply:
That's true, but I find it better readable with the additional `else`